### ISSUE TITLE
Minor Cosmetic suggestion - remove Dragoon Leap cast animation.

### ISSUE THREAD

**Issue by SebastianPop** [NosVladimir/KineticArchetypes#33]
This isn't really a bug or glitch, but something that bothers me enough that I just can't bring myself to play Kinetic Lancer, even though the archetype is the one and only reason I have this mod installed - I really dislike the magical casting animation before the jump happens. Is it possible to change it to something else? Or just remove it completely? Maybe it can be switched to the gryphon or hippogriff dive ability instead, or just a frozen two-handed charge stance during the jump?

Is there a file I can edit to do this myself so you don't have to bother with my trivial personal preferences?

**Comment by NosVladimir** [comment]
Thanks for the suggestion. The animation for gryphon or hippogriff cannot be directly applied to a humanoid so it's impossible to use those for the dragoon dive. I'm thinking of implementing a second button for dragoon dive with everything the same except with no animations whatsoever, while keeping the original button still available, with the mind that this may be helpful for the other non-attacking issue

**Comment by SebastianPop** [comment]
Thanks for responding so promptly! I'll certainly prefer to have the leap happen immediately, though I'll reiterate the idea of using the two-handed charge run under 'paralyzed' frame-freeze DURING the leap.

As for the griffon leap ability, does that mean it's impossible to apply the animation to the humanoid paperdoll (character model), or the racial parameter? Because then you could just include a shapeshift effect just for the duration of the ability, except the appearance of the character doesn't change. I suppose that might break other things, though. Or could be too bothersome to implement, since it's basically two abilities stacked on top of each other: shapeshift without visual change + leap attack.

Anyway, yes, the lack of a jump animation is a real shame. Doubt it can be imported from one game to the other either, even if they both are 3D, technically. Certainly beyond the scope of your mod.

**Comment by NosVladimir** [comment]
A frozen charge animation definitely sounds good. I'll look into that but no guarantees.

For the griffon leap, I can't say anything about how the animation works with the griffon model because I have absolutely zero experience in animations both in the game and outside of the game. The impossibility of using the griffon animation for humanoid is heard from other mod devs, from a pretty long time ago. The idea of shapeshift w/o visual change + leap attack might possibly work, but I imagine even it would, it's highly likely the character might just idle - disappear - reappear while being idle - attack. If it worked well I'd include that in the next update.

**Comment by SebastianPop** [comment]
Thanks, and Merry Christmas! 

(now that I think about it, I could probably have let this lie for a couple more days, my bad).

**Comment by NosVladimir** [comment]
Merry Christmas too!

No worries, I don't have many other things to do in this holiday time. That's also why I responded even more quickly than I normally would, lol

**Comment by NosVladimir** [comment]
Unfortunately, despite many tries I was not able to get the desired frozen charge animation working, not even the charge itself.

It looks like charge is just plain movement (running) with weapon drawn at hand, and movement is handled by a complicated system I cannot understand, in which the animation is embedded, and I can't extract the animation out of it then apply to the character on-demand.

I've added a funny vertical jump style, but it's again lacking appropiate animation. It's just the model flys up, then flys down

**Comment by SebastianPop** [comment]
Well, I appreciate the effort regardless. You didn't need to work on it during Christmas though.

**Comment by NosVladimir** [comment]
Thanks for your kind words, but in my home country we don't actually celebrate Christmas as much - though where I live atm does and hence I have holiday - guess where I'm from :)

Which Flying Kick are you talking about? Is it from a mod?

**Comment by SebastianPop** [comment]
If it's not canon, I'm not actually sure? It's basically the Liu kang flying kick, and yes it does have its own animation. A bit janky if the jump is short, but the monk does bend sideways and kick mid-air more or less like you'd expect. I'll look through the mod readmes to see and get back to you.

**Comment by SebastianPop** [comment]
Ok, I've looked at it and the Flying Kick Style Strike is from the mod Microscopic Content Expansion.

**Comment by NosVladimir** [comment]
Thanks, that's a good learning source. Separate story, but I actually talked with the author of MCE for implementation of dragoon dive attack and we agreed on the fact that Owlcat code is just being very unhappy on attacks that's not initiated by a standard attack, or attacking ability.

The problem with not using the magic casting animation is, if I don't use it, then it would end up having no animation at all, like the vertical jump I just added. This is list of animations I can easily reference and use:

`
{
    Unused = -1,
    None = 0,
    LocoMotion = 1,
    MainHandAttack = 2,
    MainHandEquip = 3,
    MainHandUnequip = 4,
    OffHandAttack = 5,
    OffHandEquip = 6,
    OffHandUnequip = 7,
    Prone = 8,
    StandUp = 9,
    SpecialAttack = 10,
    CastSpell = 11,
    Interact = 12,
    MicroIdle = 13,
    Dodge = 14,
    Hit = 0xF,
    Ragdoll = 0x10,
    DialogCueAnimation = 18,
    VariantIdle = 21,
    Stunned = 23
}
`
where `LocoMotion` includes all animation for different types movements (slow, normal, fast) and is handled by the aforementioned complicated system, and these are the options for `CastSpell`:
`
SelfTouch,
Touch,
Point,
Directional,
Omni,
Self,
Thrown,
Potion,
Special,
Immediate,
EnchantWeapon,
BreathWeapon,
Kineticist,
CoupDeGrace,
MagicHack
`
IMO nothing fits into the jump unless I develop a new animation. The `CastSpell` animation with `Kineticist` style is the best among all these options already...

**Comment by SebastianPop** [comment]
That's a shame. I thought it would at least be possible to apply the 'instant spell' metamagic or something, but if you say the animation just flat out doesn't happen, guess it's just another case of 'Unity strikes again.' If all that could be tried has been tried, I must but cope. Thanks for trying anyway.

**Comment by NosVladimir** [comment]
Looking at the code (https://github.com/alterasc/MicroscopicContentExpansion/blob/main/MicroscopicContentExpansion/NewComponents/SpecialAnimationMonkKick.cs) it seems it's just referencing a `MainHandAttack` with `WeaponAnimationStyle.MartialArts` and `Variant = 2`. It could be that there is already a side kick animation for unarmed strikes. That's a clever reference use!

**Comment by SebastianPop** [comment]
There is, the monk has two different fist attacks, and two kick attacks, one of which is a spin kick. I'm more interested in how the MCE flying kick doesn't use any pre-cast animations at all, and just uses the kick one. Maybe he make the kick animation count as the casting animation?

I don't know anything about Unity modding, so ignore me if I'm saying shit you've already accounted for.

**Comment by NosVladimir** [comment]
I'm not farmilar with monk attacks and can't say for sure, but I guess that's what the `Variant = 2` does - straight asking for the kick animation

**Comment by SebastianPop** [comment]
Theoretically, it should work the same way for any attack animation? This could solve your other reported issue, at least, and make it look like a pretty slick pounce attack with the kinetic spear.

Doesn't do anything for the raw kick though, unfortunately.

**Comment by NosVladimir** [comment]
True, I'd have to double think about that and take a closer look at MCE's code, and also see what the flying kick actually looks like. Maybe I'll just have another chat with alterasc

**Comment by SebastianPop** [comment]
Shouldn't be too big an issue for the latter - all weapons have a thrust animation, it won't look strange if you just use the longspear one, the game allows you to use one-handed weapons with two hands after all.

For the jump though, I got nothing. Unless the code actually lets you play attack animations for things other than attacking, you'd be left with, what? Stealth crouch? But the charge is just movement animation, and you say it doesn't work outside a charge action, so that probably won't work either?

**Comment by NosVladimir** [comment]
> Unless the code actually lets you play attack animations for things other than attacking

 Exactly this. The code lets me play whichever animation I want, at any time, except that one movement/locomotion/running/charging/whateveritscalled animation.

This has been what I and alterasc was doing for our mods. I set the actual casting animation of dragoon dive to none, and play my own `CastSpell - Kineticist` animation to mimic a jump, then perform actual attacks with proper attacking animations. Flying kick is even more clever, it has no actual casting animation, and plays `MainHandAttack` animation at some time, and triggers `RuleAttackWithWeapon` at the end to mimic the flying kick attack. `RuleAttackWithWeapon` involves no animation, you can think of it as the pesudo-attack when you use cleaving finish to cut through multiple enemies - nearby enemies are hurt, but you don't have the animation.
